llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -99,6 +100,13 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
   void computeMaxLatency();
 
 public:
+  enum class ConflictType : std::uint32_t {
+    NoConflict = 0b000,
+    Format = 0b001,
+    MemoryBank = 0b010,
+    FuncUnit = 0b100,
+  };
```

Comment:
Nit: To avoid the constant `static_cast`, I think we could create e.g.
```
class ConflictTypeSet {
uint32_t BitSet = 0b0;
static uint32_t  asBit(ConflictType CT) { return uint32_t(CT); }
public:
  bool has(ConflictType CT) const { return BitSet & asBit(CT); }
  bool any() const { return BitSet != 0b0; }
  void set(ConflictType CT) { BitSet |= asBit(CT); }
};